woocommerce/woocommerce-ios
# Simplified login: Apply A/B test

Closes: #7904 


### Description

This PR adds a new A/B test experiment for simplified login to enable only WP.com logins. Experiment details: pbxNRc-27s-p2.
Changes:
- Added new A/B test experiment.
- Replaced the temporary feature flag with A/B test.
- Removed the unnecessary checks on password screen to use the new design on all variants.
- Removed the unnecessary checks on unrecognized email error screen to use the new design on all variants.

### Testing instructions


#### `control` variation

This is the default variation, and most likely the one that is shown before the experiment is launched.

- Delete the app
- Launch the app from the PR branch --> the onboarding screen should be shown, and an event is logged `🔵 Tracked application_installed, properties: [AnyHashable("after_abtest_setup"): true]`
- Finish or dismiss the onboarding --> the prologue screen should be shown with entry points to both WP.com login and site address login.
- Tap "Continue with WordPress.com", notice the email screen has entry points to social logins and a link for "What is WordPress.com".

#### Testing the `treatment` variation

It's difficult to test different variants in A/B tests because ExPlat is supposed to assign one single variant to a device. I'll shamelessly paste the guide from Jaclyn's past PR #7551 here:

Prerequisites:
- Please follow the setup in PCYsg-Fq7-p2#assignments-api for proxied sandboxed API requests: connect to sandbox via ssh, and point WP.com API to the sandbox IP address (`ifconfig`) in the local matchine's `/etc/hosts`
- In Xcode, set a breakpoint at `ExPlatService` L47 where `anon_id` is set in the URL request
- Make an API request `PATCH /wpcom/v2/experiments/0.1.0/assignments` with the following `body` and make sure it's successful:
```
{
    "variations": {
        "woocommerceios_login_wpcom_only": "treatment"
    },
    "anon_id": "anon_id_from_the_previous_step",
    "username_override": ""
}
```
- Launch the app, and let it sit for a bit
- Relaunch the app --> the treatment variation should be shown with only the CTA "Log In".
- Tap "Log In", notice that the email screen is the simplified design with no mentioning of WordPress.com.

### Screenshots

| Control | Treatment |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/199221300-16caa188-ec31-4975-a27d-accd10e5da25.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/199221377-8a0278ce-03bd-4694-be39-0f8b050df8e4.png" width=320 /> | 
| <img src="https://user-images.githubusercontent.com/5533851/199221523-8e95d602-12f0-4a1d-b104-a009412a4fa7.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/199221605-38998c22-95c9-4298-866f-5f83402747dd.png" width=320 /> |




---
